pkg/storagev2/leveldb/store.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package leveldbstore
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	storageV2 "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/hashicorp/go-multierror"
+	"github.com/syndtr/goleveldb/leveldb"
+	ldbErrors "github.com/syndtr/goleveldb/leveldb/errors"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/storage"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type Store struct {
+	DB      *leveldb.DB
+	path    string
+	closeLk sync.RWMutex
+}
+
+var _ storageV2.Store = (*Store)(nil)
+
+// NewDatastore returns a new datastore backed by leveldb
+// for path == "", an in memory backend will be chosen (TODO)
+func New(path string, opts *opt.Options) (storageV2.Store, error) {
+	var err error
+	var db *leveldb.DB
+
+	if path == "" {
+		db, err = leveldb.Open(storage.NewMemStorage(), opts)
+	} else {
+		db, err = leveldb.OpenFile(path, opts)
+		if ldbErrors.IsCorrupted(err) && !opts.GetReadOnly() {
+			db, err = leveldb.RecoverFile(path, opts)
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	ds := Store{
+		DB:   db,
+		path: path,
+	}
+
+	return &ds, nil
+}
+
+func (s *Store) Count(key storageV2.Key) (int, error) {
+	s.closeLk.RLock()
+	defer s.closeLk.RUnlock()
+
+	keys := util.BytesPrefix([]byte(key.Namespace() + "/"))
```

Comment:
So I would make the `/` constant. It is highly unlikely but its easier if we want to change this in future.

Reply:
fixed